extensions/2.0/Khronos/KHR_materials_variants/schema/glTF.KHR_materials_variants.schema.json

```diff
@@ -9,6 +9,7 @@
             "type": "array",
             "items": {
                 "type": "object",
+                "allOf": [ { "$ref": "glTFChildOfRootProperty.schema.json" } ],
```

Comment:
For consistency: https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_lights_punctual/schema/glTF.KHR_lights_punctual.schema.json defines this differently:

```
"items": {
                "type": "object",
                "$ref": "light.schema.json"
},
```

... are these equivalent?

Reply:
`$ref` would require adding another file like `variant.schema.json`. Given that we don't need any fields beyond those defined in `glTFChildOfRootProperty`, I think taking a shortcut here is fine.

Reply:
I didn't mean to suggest creating a new file — comparing these two lines:

```
"$ref": "glTFChildOfRootProperty.schema.json",
```

```
"allOf": [ { "$ref": "glTFChildOfRootProperty.schema.json" } ],
``` 

... is there any difference in meaning? Or why does the first require a new file when the second does not?

Reply:
>  is there any difference in meaning?

The semantics are slightly different.

This case
```json
"items": {
  "type": "object",
  "$ref": "glTFChildOfRootProperty.schema.json"
},
```
means: "replace the value of `"items"` with the contents of `"glTFChildOfRootProperty.schema.json"`". In fact, `"type": "object"` is ignored there.

OTOH, this case
```json
"items": {
  "type": "object",
  "allOf": [ { "$ref": "glTFChildOfRootProperty.schema.json" } ],
```
means: "`items` is an object that must be valid against all the schemas defined in `allOf` array".

Given that `KHR_materials_variants` defines `name` as a required field, we cannot use the first approach here.

Reply:
I see, thanks!